src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -377,13 +377,30 @@ def cleanRSE(self, rse):
                             self.logger.exception(msg, rse['name'], str(ex))
 
                     self.logger.info("RSE: %s, Dir: %s, filesDeletedSuccess: %s",
-                                      rse['name'], dirLfn, deletedSuccess)
-                    rse['counters']['filesDeletedSuccess'] += deletedSuccess
-
-                    # Now clean the whole branch
-                    self.logger.debug("Purging dirEntry: %s:\n", dirPfn)
-                    purgeSuccess = self._purgeTree(ctx, dirPfn)
-                    if purgeSuccess:
+                                      rse['name'], dirLfn, filesDeletedSuccess)
+                    rse['counters']['filesDeletedSuccess'] += filesDeletedSuccess
+
+                    # Now delete the whole branch
+                    # First try to delete the base directory:
+                    rmdirSuccess = False
+                    purgeSuccess = False
+                    try:
+                        self.logger.debug("Trying to remove Dir: %s", dirPfn)
+                        # NOTE: For gfal2 rmdir() exit status of 0 is success
+                        rmdirSuccess = ctx.rmdir(dirPfn) == 0
+                    except gfal2.GError as gfalExc:
+                        if gfalExc.code == errno.ENOENT:
+                            self.logger.warning("MISSING Dir: %s", dirPfn)
+                            rmdirSuccess = True
+                        else:
+                            self.logger.error("FAILED to remove Dir: %s: gfalException: %s", dirPfn, str(gfalExc))
```

Comment:
Maybe make this error log more user friendly instead of developer friendly? E.g.: `FAILED to remove directory: %s . GFAL exception: %s"

Reply:
done